runtime-modules/working-team/src/lib.rs

```diff
@@ -695,4 +918,115 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
         let slashed_balance = T::StakingHandler::slash(staking_account_id, balance);
         Self::deposit_event(RawEvent::StakeSlashed(worker_id, slashed_balance));
     }
+
+    // Reward a worker using reward presets and working team budget.
+    fn reward_worker(worker_id: &TeamWorkerId<T>, worker: &TeamWorker<T>) {
+        // If reward period is not set.
+        let mut rewarding_period: u32 = T::RewardPeriod::get();
+        if rewarding_period == Zero::zero() {
+            rewarding_period = One::one();
+        }
+
+        // Modify rewarding period for new workers.
+        let block_from_worker_creation: u128 =
+            (Self::current_block() - worker.created_at).saturated_into();
+        if block_from_worker_creation < rewarding_period.into() {
+            rewarding_period = block_from_worker_creation.saturated_into();
+        }
+
+        if let Some(reward_per_block) = worker.reward_per_block {
+            let reward = reward_per_block * rewarding_period.into();
+
+            let budget = Self::budget();
+
+            // Pay the reward.
+            if budget >= reward {
+                let new_budget = budget - reward;
+                <Budget<T, I>>::put(new_budget);
+
+                T::Currency::deposit_creating(&worker.reward_account_id, reward);
+
+                Self::try_to_pay_missed_reward(worker_id, worker);
+            } else {
+                // Save unpaid reward.
+                let missed_reward_so_far = worker.missed_reward.map_or(Zero::zero(), |val| val);
+
+                let new_missed_reward = missed_reward_so_far + reward;
+
+                // Update worker missed reward.
+                WorkerById::<T, I>::mutate(worker_id, |worker| {
+                    worker.missed_reward = Some(new_missed_reward);
+                });
+            }
+        }
+    }
+
+    // Tries to pay missed reward if the reward is enabled for worker and there is enough of team budget.
+    fn try_to_pay_missed_reward(worker_id: &TeamWorkerId<T>, worker: &TeamWorker<T>) {
```

Comment:
I think it makes sense to try to pay as much as you can, rather than making it all or nothing. For example, if a worker is leaving, it does not make sense to have them lose _all_ their owed money, just because you could not pay some of it. I would say that applies to normal payouts also. It also discouraged debts from growing extremely large to some extent.

Reply:
Done.